src/test/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingTest.java

```diff
@@ -74,6 +74,226 @@ public void TranslateTriplePatternTest() {
 		assertEquals(testData.object2, translationTriples);
 	}
 	
+	@Test
+	public void TranslateSubjectTest() {
+		final Set<Triple> testTriples = new HashSet<>();
+		//Equality
+		Node s = NodeFactory.createLiteral("s1");
+		Node p = OWL.sameAs.asNode();
+		final Node o1 = NodeFactory.createLiteral("s2");
+		testTriples.add(new Triple(s, p, o1));
+				
+		//Multiple mappings for same subject
+		final Node o2 = NodeFactory.createLiteral("s3");
+		testTriples.add(new Triple(s, p, o2));
+		
+		final VocabularyMapping vm = new VocabularyMappingImpl(testTriples);
+		
+		p = NodeFactory.createLiteral("p");
+		final Node o = NodeFactory.createLiteral("o");
+		final TriplePattern testTp = new TriplePatternImpl(s, p, o);
+		final SPARQLGraphPattern translation = vm.translateTriplePattern(testTp); 
+		final Set<Triple> translationTriples = new HashSet<>();
+		assertTrue(translation instanceof SPARQLUnionPatternImpl);
+		for (final SPARQLGraphPattern i : ((SPARQLUnionPatternImpl) translation).getSubPatterns()) {
+			assertTrue(i instanceof TriplePattern);
+			translationTriples.add(((TriplePattern) i).asJenaTriple());
+		}
+			
+		final Set<Triple> expectedResults = new HashSet<>();
+		expectedResults.add(new Triple(o1, p, o));
+		expectedResults.add(new Triple(o2, p, o));
+		
+		assertEquals(translationTriples, expectedResults);
```

Comment:
The expected results should go first. This is because in case the test fails, it will show you the differences correctly.

```suggestion
		assertEquals(expectedResults, translationTriples);
```